### PR TITLE
TIP-560: standardirze readers/processors/writers

### PR DESCRIPTION
**Pull request by jjanvier**
What's done here:
* all readers/processors/writers class names have been standardized (no `Simple` or `Base` prefix anymore)
* all readers/processors/writers service names have been standardized
* `AbstractConfigurableStepElement` has been removed
* all readers/processors/writers now implement the required interfaces in their concrete implementation (not only in the abstract or parent class)

Checklist:
 * [x] readers
 * [x] processors 
 * [x] writers
 * [x] check all AbstractConfigurableStepElement
 * [x] check all StepExecutionAwareInterface

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | -
| Added Behats                      | -
| Changelog updated                 | -
| Review and 2 GTM                  |
| Micro Demo to the PO (Story only) | -
| Migration script                  | -
| Tech Doc                          | -

### CODE REVIEW

File: src/Pim/Component/Connector/Processor/Denormalization/JobInstanceProcessor.php [hunk 1]
```diff
@@ -20,7 +22,7 @@
  * @copyright 2016 Akeneo SAS (http://www.akeneo.com)
  * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
  */
-class JobInstanceProcessor extends AbstractProcessor
+class JobInstanceProcessor extends AbstractProcessor implements ItemProcessorInterface, StepExecutionAwareInterface
```
Comment (skeleton) [inline src/Pim/Component/Connector/Processor/Denormalization/JobInstanceProcessor.php]: remove ItemProcessorInterface

**Comment by skeleton** [comment]
fix classname of `spec\Pim\Component\Connector\Reader\Database\ReaderSpec` please, then GTM

File: src/Pim/Component/Connector/Processor/Denormalization/JobInstanceProcessor.php (hunk 1 above)
Comment (jjanvier) [inline src/Pim/Component/Connector/Processor/Denormalization/JobInstanceProcessor.php]: done in the abstract

**Comment by jjanvier** [comment]
@skeleton done

File: src/Akeneo/Component/Batch/Item/AbstractConfigurableStepElement.php [hunk 2]
```diff
@@ -1,31 +0,0 @@
...
-
-use Akeneo\Component\Batch\Step\StepElementInterface;
-
-/**
- * Define a configurable step element
- *
- * @copyright 2016 Akeneo SAS (http://www.akeneo.com)
- * @license   http://opensource.org/licenses/MIT MIT
- */
-abstract class AbstractConfigurableStepElement implements
```
Comment (nidup) [inline src/Akeneo/Component/Batch/Item/AbstractConfigurableStepElement.php]: :dancer: :fire:

File: src/Pim/Bundle/ConnectorBundle/Resources/config/readers.yml [hunk 3]
```diff
@@ -91,7 +91,7 @@ services:
         class: '%pim_connector.reader.dummy_item.class%'
 
     # CSV File Readers
-    pim_connector.reader.file.csv:
+    pim_connector.reader.file.csv_dummy:
```
Comment (nidup) [inline src/Pim/Bundle/ConnectorBundle/Resources/config/readers.yml]: weird, why csv_dummy?

**Comment by nidup** [comment]
don't understand the csv_dummy stuff, GTm once understood (same comment in EE PR)

File: src/Pim/Component/Connector/Writer/File/Xlsx/VariantGroupWriter.php [hunk 4]
```diff
@@ -13,7 +16,12 @@
  * @copyright 2016 Akeneo SAS (http://www.akeneo.com)
  * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
  */
-class VariantGroupWriter extends AbstractItemMediaWriter implements ItemWriterInterface, ArchivableWriterInterface
+class VariantGroupWriter extends AbstractItemMediaWriter implements
+    ItemWriterInterface,
+    InitializableInterface,
+    FlushableInterface,
```
Comment (juliensnz) [inline src/Pim/Component/Connector/Writer/File/Xlsx/VariantGroupWriter.php]: could you please unalign those ?

**Comment by juliensnz** [comment]
Not GTM :trollface:

File: src/Pim/Bundle/ConnectorBundle/Resources/config/readers.yml (hunk 3 above)
Comment (jjanvier) [inline src/Pim/Bundle/ConnectorBundle/Resources/config/readers.yml]: ok seen IRL, this one is deleted, and we create the dummy services where we need it in EE

File: src/Pim/Component/Connector/Writer/File/Xlsx/VariantGroupWriter.php (hunk 4 above)
Comment (jjanvier) [inline src/Pim/Component/Connector/Writer/File/Xlsx/VariantGroupWriter.php]: arent't they?








meurs...

**Pull request merged**